app/models/specific_tube_creation.rb

```diff
@@ -64,13 +64,41 @@ def connect_parent_and_children
   end
   private :connect_parent_and_children
 
+  def determine_foreign_barcode_format(foreign_barcode)
+    Barcode.matching_barcode_format(foreign_barcode)
```

Comment:
Would it make more sense to explicitly specify the barcode format? Rather than trying to auto-detect it. Presumably we know what kind of tubes we're registering.